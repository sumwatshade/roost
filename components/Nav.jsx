import React from 'react';
import NextMuiLink from './NextMuiLink';
import {Typography, AppBar, Toolbar} from '@mui/material';
import {styled} from '@mui/material/styles';

const routes = [{
  name: 'Home',
  path: '/',
}, {
  name: 'About',
  path: '/about',
}];

const RouteTypography = styled(Typography)(({theme}) => ({
  marginLeft: theme.spacing(3),
  color: 'white',
}));

const Nav = () => {
  return (
    <AppBar position='static' component='nav'>
      <Toolbar>
        {routes.map(({name, path}) => {
          return (<NextMuiLink key={name} href={path} color='secondary'>
            <RouteTypography variant="h6">
              {name}
            </RouteTypography>
          </NextMuiLink>);
        })}
      </Toolbar>
      <style jsx>{`
        .navigation {
            display: flex;
            flex-direction: row;
            margin: 0 5%;
            width: 40%;
            height: 60px;
            align-items: center;
            justify-content: space-evenly

        }

        @media(max-width: 480px) {
          .navigation {
            width: 100%;
            margin: 0;
          }
        }
      `}</style>
    </AppBar>
  );
};

export default Nav;
