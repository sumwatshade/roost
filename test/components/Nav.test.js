import React from "react";
import { render } from "@testing-library/react";

import Nav from "../../components/Nav";

jest.mock("../../components/NextMuiLink");

describe("Nav Component", () => {
  test("renders in default state", () => {
    const { asFragment } = render(<Nav />);

    expect(asFragment()).toMatchSnapshot();
  });
});
