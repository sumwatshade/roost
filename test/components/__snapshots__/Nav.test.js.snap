// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component renders in default state 1`] = `
<DocumentFragment>
  <nav
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1vrl3mg-MuiTypography-root-MuiLink-root"
        href="/"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-io3amt-MuiTypography-root"
        >
          Home
        </h6>
      </a>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1vrl3mg-MuiTypography-root-MuiLink-root"
        href="/about"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-io3amt-MuiTypography-root"
        >
          About
        </h6>
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
